packages/app-store/routing-forms/components/SingleForm.tsx

```diff
@@ -229,33 +230,145 @@ type SingleFormComponentProps = {
+      (member) => member.email
+    );
+
+    if (matchingMembers.length) {
+      return <span className="font-semibold">{matchingMembers.join(", ")}</span>;
+    }
+
+    return (
+      <span className="font-semibold">
+        {t("all_assigned_members_of_the_team_event_type_consider_tweaking_fallback_to_have_a_match")}
+      </span>
+    );
+  };
+
+  return (
+    <div className="text-default mt-2 space-y-2">
+      <div data-testid="chosen-route">
+        {t("chosen_route")}: <span className="font-semibold">{chosenRouteName}</span>
+      </div>
+      <div data-testid="attribute-logic-matched" className={cn(hasMainWarnings && "text-error")}>
+        {t("attribute_logic_matched")}: <span className="font-semibold">{renderMainLogicStatus()}</span>
+        {hasMainWarnings && (
+          <Alert className="mt-2" severity="warning" title={membersMatchResult.mainWarnings?.join(", ")} />
+        )}
+      </div>
+      <div data-testid="attribute-logic-fallback-matched" className={cn(hasFallbackWarnings && "text-error")}>
+        {t("attribute_logic_fallback_matched")}:{" "}
+        <span className="font-semibold">{renderFallbackLogicStatus()}</span>
+        {hasFallbackWarnings && (
+          <Alert
+            className="mt-2"
+            severity="warning"
+            title={membersMatchResult.fallbackWarnings?.join(", ")}
+          />
+        )}
+      </div>
+      <div data-testid="matching-members">
+        {t("matching_members")}: {renderMatchingMembers()}
+      </div>
+    </div>
+  );
+
+  function isNoLogicFound(
+    teamMembersMatchingAttributeLogic: NonNullable<MembersMatchResultType>["teamMembersMatchingAttributeLogic"]
+  ): teamMembersMatchingAttributeLogic is null {
+    return teamMembersMatchingAttributeLogic === null;
+  }
+};
+
+export const TestFormDialog = ({
+  form,
+  isTestPreviewOpen,
+  setIsTestPreviewOpen,
+}: {
+  form: NonNullable<SingleFormComponentProps["enrichedWithUserProfileForm"]>;
+  isTestPreviewOpen: boolean;
+  setIsTestPreviewOpen: (value: boolean) => void;
+}) => {
   const { t } = useLocale();
-  const isTeamForm = !!form.teamId;
-  const [isTestPreviewOpen, setIsTestPreviewOpen] = useState(false);
   const [response, setResponse] = useState<FormResponse>({});
   const [chosenRoute, setChosenRoute] = useState<NonRouterRoute | null>(null);
-  const [skipFirstUpdate, setSkipFirstUpdate] = useState(true);
   const [eventTypeUrl, setEventTypeUrl] = useState("");
   const searchParams = useCompatSearchParams();
-  const [teamMembersMatchingAttributeLogic, setTeamMembersMatchingAttributeLogic] = useState<
-    | {
-        id: number;
-        name: string | null;
-        email: string;
-      }[]
-    | null
-  >([]);
+  const isTeamForm = !!form.teamId;
+  const [membersMatchResult, setMembersMatchResult] = useState<MembersMatchResultType | null>(null);
+
+  const resetMembersMatchResult = () => {
+    setMembersMatchResult(null);
```

Comment:
@CarinaWolli Now setting the correct state value which just says that we don't have anhy data.